src/toil_scripts/adam_kmers/count_kmers.py

```diff
@@ -0,0 +1,234 @@
+
+        # change the file extension to adam
+        hdfs_input_file = '.'.join(hdfs_input_file.split('.')[:-1] + ['adam'])
+
+        # convert the file
+        _log.info('Converting %s into ADAM format at %s.', hdfs_tmp_file, hdfs_input_file)
+        call_adam(master_ip,
+                  ['transform',
+                   hdfs_tmp_file, hdfs_input_file],
+                  memory=memory, override_parameters=spark_conf)
+        
+    # run k-mer counting
+    _log.info('Counting %d-mers in %s, and saving to %s.',
+              kmer_length, hdfs_input_file, hdfs_output_file)
+    start_time = time.clock()
+    call_adam(master_ip,
+              ['count_kmers',
+               hdfs_input_file, hdfs_output_file,
+               str(kmer_length)],
+              memory=memory, override_parameters=spark_conf)
+    end_time = time.clock()
+    job.fileStore.logToMaster('Run time of ADAM job was %f seconds.' % (end_time - start_time))
+
+    # do we need to upload the file back? if so, run upload
+    if run_upload:
+        _log.info("Uploading output file %s to %s.", hdfs_output_file, output_file)
+        call_conductor(master_ip, hdfs_output_file, output_file,
+                       memory=memory, override_parameters=spark_conf)
+        
+
+def main():
+    '''
+    Sets up command line parser for Toil/ADAM based k-mer counter, and launches
+    k-mer counter with optional Spark cluster.
+    '''
+
+    parser = argparse.ArgumentParser()
+
+    # add parser arguments
+    parser.add_argument('--input-file',
+                        help='The full path to the input SAM/BAM/ADAM/FASTQ file.')
+    parser.add_argument('--output-file',
+                        help='full path where final results will be output.')
+    parser.add_argument('--kmer-length',
+                        help='Length to use for k-mer counting. Defaults to 20.',
+                        default=20,
+                        type=int)
+    parser.add_argument('--spark-conf',
+                        help='Optional configuration to pass to Spark commands. Either this or --workers must be specified.',
+                        default=None)
+    parser.add_argument('--memory',
+                        help='Optional memory configuration for Spark workers/driver. This must be specified if --workers is specified.',
+                        default=None,
+                        type=int)
+    parser.add_argument('--cores',
+                        help='Optional core configuration for Spark workers/driver. This must be specified if --workers is specified.',
+                        default=None,
+                        type=int)
+    parser.add_argument('--workers',
+                        help='Number of workers to spin up in Toil. Either this or --spark-conf must be specified. If this is specified, --memory and --cores must be specified.',
+                        default=None,
+                        type=int)
+    parser.add_argument('--sudo',
```

Comment:
`spawn_spark_cluster` does, though.
We should unify the two. Why did we remove `sudo` from `docker_call`?